src/snowflake/connector/direct_file_operation_utils.py

```diff
@@ -49,12 +81,37 @@ def parse_file_operation(
         options,
         has_source_from_stream=False,
     ):
-        raise NotImplementedError("parse_file_operation is not yet supported")
+        """Parses a file operation by constructing SQL and getting the SQL parsing result from server."""
+        options_in_sql, option_bind_values = self._process_options_for_upload(options)
+
+        if command_type == CMD_TYPE_UPLOAD:
```

Comment:
This does not look like a pure refactor. We do an assertion, a file name split and some escaping. These do not exist in the original code. Why are they required?